usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -3388,8 +3472,17 @@ metaslab_check_free(spa_t *spa, const blkptr_t *bp)
 		uint64_t size = DVA_GET_ASIZE(&bp->blk_dva[i]);
 		metaslab_t *msp = vd->vdev_ms[offset >> vd->vdev_ms_shift];
 
-		if (msp->ms_loaded)
+		if (msp->ms_loaded) {
 			range_tree_verify(msp->ms_tree, offset, size);
+#ifdef	DEBUG
+			range_tree_verify(msp->ms_cur_ts->ts_tree,
+			    offset, size);
+			if (msp->ms_prev_ts != NULL) {
+				range_tree_verify(msp->ms_prev_ts->ts_tree,
+				    offset, size);
+			}
+#endif
+		}
```

Comment:
Seems it's not in there yet. I'll modify this PR to include it and see if there's any merge problems later.